pallets/committee-management/src/tests.rs

```diff
@@ -0,0 +1,204 @@
+        non_reserved_finality_seats: 4,
+    }
+}
+
+#[test]
+fn new_poducers_every_session() {
+    TestExtBuilder::new(gen_config()).build().execute_with(|| {
+        let mut producers_in_all_sessions = BTreeSet::<BTreeSet<AccountId>>::new();
+        for session_index in 2..=6 {
+            start_session(session_index);
+            let producers = CommitteeManagement::current_session_validators()
+                .current
+                .producers
+                .into_iter()
+                .collect();
+
+            assert!(producers_in_all_sessions.insert(producers));
+        }
+    })
+}
+
+#[test]
+fn new_finalizers_every_session() {
+    TestExtBuilder::new(gen_config()).build().execute_with(|| {
+        let mut finalizers_in_all_sessions = BTreeSet::<BTreeSet<AccountId>>::new();
+        for session_index in 2..=6 {
+            start_session(session_index);
+            let finalizers = CommitteeManagement::current_session_validators()
+                .current
+                .finalizers
+                .into_iter()
+                .collect();
+            assert!(finalizers_in_all_sessions.insert(finalizers));
+        }
+    })
+}
+
+#[test]
+fn all_reserved_validators_are_chosen() {
+    TestExtBuilder::new(gen_config()).build().execute_with(|| {
+        let reserved = Elections::current_era_validators().reserved;
+        start_session(2);
+        let producers: BTreeSet<AccountId> = CommitteeManagement::current_session_validators()
+            .current
+            .producers
+            .into_iter()
+            .collect();
+        assert!(reserved.iter().all(|rv| producers.contains(rv)));
+        let finalizers: BTreeSet<AccountId> = CommitteeManagement::current_session_validators()
+            .current
+            .finalizers
+            .into_iter()
+            .collect();
+        assert!(reserved.iter().all(|rv| finalizers.contains(rv)));
+    })
+}
+
+#[test]
+fn ban_underperforming_producers() {
+    TestExtBuilder::new(gen_config()).build().execute_with(|| {
+        let underperformer = 10;
+        let mut ban_config = CommitteeManagement::producers_ban_config();
+        let underperformed_session_count_threshold =
+            ban_config.underperformed_session_count_threshold;
+        let reserved: BTreeSet<AccountId> = Elections::current_era_validators()
+            .reserved
+            .into_iter()
+            .collect();
+        let blocks_to_produce_per_session = SessionPeriod::get();
+        let mut underperf_count = 0;
+        let mut session_index = 2;
+        loop {
+            start_session(session_index);
+            if underperf_count == underperformed_session_count_threshold {
+                break;
+            }
+            assert_eq!(CommitteeManagement::banned(), Vec::<AccountId>::new());
+
+            // Make sure underperformer is a producer in every session.
+            let producers = CurrentAndNextSessionValidatorsStorage::<TestRuntime>::mutate(|sv| {
```

Comment:
This test is a little long in terms of code lines, maybe we can extract some code e.g. this seems like a good method candidate like `add_underperformer_to_committee`, which we can re-use in `ban_underperforming_finalizers` test.

Reply:
done